### PR TITLE
Create new iModels with WAL mode enabled

### PR DESCRIPTION
**Pull request by kabentley**

### CODE REVIEW

File: iModelCore/iModelPlatform/DgnCore/DgnDb.cpp [hunk 1]
```diff
@@ -665,100 +645,93 @@ DgnDbPtr DgnDb::OpenDgnDb(DbResult* outResult, BeFileNameCR fileName, OpenParams
...
+    }
 
-    DbResult rc = CreateNewDb(projectFile, params, params.GetGuid());
+    DbResult rc = CreateNewDb(iModelName, params, params.GetGuid());
     if (BE_SQLITE_OK != rc)
         return rc;
 
-    m_fileName = projectFile.GetNameUtf8();
+    // iModels should always use WAL journal mode
+    if (!memoryDb)
```
Comment (pmconne) [inline iModelCore/iModelPlatform/DgnCore/DgnDb.cpp:683]: Is this the only actual functional change? Way too much unrelated rename/formatting diff to sift through.

Comment (kabentley) [reply iModelCore/iModelPlatform/DgnCore/DgnDb.cpp:683]: yes, the call to EnableWalMode is new and the object of the PR. Several tests broke because they attempted to copy an open file which doesn't work in WAL mode unless you checkpoint first. That precipitated the renames so I could find all callers of create/open.

**Review by pmconne — APPROVED** [review]
